### PR TITLE
fix-for-issue-7691

### PR DESCRIPTION
**Pull request by shashwat70**
Issue #7691 : update doc for ClassFanOutComplexity

Output of default example:

```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassFanOutComplexityCheck">
    </module>
  </module>
</module>

C:\Users\Shashwat\Documents>cat Test.java
package a.b;

import a.b.Bar;
import a.b.c.Baz;

public class Foo {
  public Bar bar; // Will be ignored, located inside ignored a.b package
  public Baz baz; // Will not be ignored, located inside a.b.c package
  public Data data; // Will not be ignored, same file

  class Data {
    public Foo foo; // Will not be ignored, same file
  }
}

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

Output of example 1:

```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassFanOutComplexity">
        <property name="max" value="2"/>
    </module>
  </module>
</module>

C:\Users\Shashwat\Documents>cat Test.java
public class InputClassComplexity {
  public Set _set = new HashSet(); // Ignored due to default excludedClasses property
  public Map _map = new HashMap(); // Ignored due to default excludedClasses property
  public List<String> _list = new ArrayList<>();
                      // Ignored due to default excludedClasses property
  public Date _date = new Date(); // Counted, 1
  public Time _time = new Time(); // Counted, 2
  public Place _place = new Place(); // Counted, 3
}

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\Shashwat\Documents\Test.java:1:1: Class Fan-Out Complexity is 3 (max allowed is 2). [ClassFanOutComplexity]
Audit done.
Checkstyle ends with 1 errors.
```

Output of Example 2:

```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassFanOutComplexity">
        <property name="excludedClasses" value="ArrayList"/>
        <property name="excludeClassesRegexps" value="^Hash.*"/>
    </module>
  </module>
</module>

C:\Users\Shashwat\Documents>cat Test.java
public class InputClassComplexity {
  public Set _set = new HashSet(); // Ignored due to default excludedClasses property
  public Map _map = new HashMap(); // Ignored due to default excludedClasses property
  public List<String> _list = new ArrayList<>();
                      // Ignored due to default excludedClasses property
  public Date _date = new Date(); // Counted, 1
  public Time _time = new Time(); // Counted, 2
  public Place _place = new Place(); // Counted, 3
}

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

Output of Example 3:

```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassFanOutComplexity">
        <property name="excludedPackages" value="a.b"/>
    </module>
  </module>
</module>

C:\Users\Shashwat\Documents>cat Test.java
package a.b;

import a.b.Bar;
import a.b.c.Baz;

public class Foo {
  public Bar bar; // Will be ignored, located inside ignored a.b package
  public Baz baz; // Will not be ignored, located inside a.b.c package
  public Data data; // Will not be ignored, same file

  class Data {
    public Foo foo; // Will not be ignored, same file
  }
}

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

### CODE REVIEW

**Comment by shashwat70** [comment]
@rnveach why did this check fail? I'm unable to see the details.

**Comment by romani** [comment]
wercker was failed as:
```
* Where:
Build file '/pipeline/source/.ci-temp/spring-integration/build.gradle' line: 19

* What went wrong:
Plugin [id: 'com.jfrog.artifactory', version: '4.14.1', apply: false] 
was not found in any of the following sources:
```

relaunched.

File: pom.xml [hunk 1]
```diff
@@ -200,7 +200,7 @@
     <projectVersion>${project.version}</projectVersion>
     <antlr4.version>4.8-1</antlr4.version>
     <maven.site.plugin.version>3.8.2</maven.site.plugin.version>
-    <maven.spotbugs.plugin.version>3.1.12.2</maven.spotbugs.plugin.version>
+    <maven.spotbugs.plugin.version>4.0.0</maven.spotbugs.plugin.version>
```
Comment (romani) [inline pom.xml]: please remove from commit all not yours changes.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java [hunk 2]
```diff
@@ -87,6 +87,44 @@
...
  * <p>
+ * To configure the check with two excluded classes {@code HashMap} and {@code HashSet}:
+ * </p>
+ * <pre>
+ * &lt;module name="ClassFanOutComplexity"&gt;
+ *   &lt;property name="excludedClasses" value="HashMap, HashSet"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * To configure the check with two regular expressions {@code ^Array.*} and {@code .*Exception$}:
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java]: all configs should be followed by code block that contains violations and lines that are compliant.
It helps a lot tohave the same code (if reasonable) and change config and show how behavior is changing.
Please do.

File: pom.xml (hunk 1 above)
Comment (shashwat70) [reply pom.xml]: Thank you for your response. Done!

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java (hunk 2 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java]: I've added the examples wherever I felt necessary. I think some configs were extremely obvious. Anyway, please have a look and let me know if something still needs to be fixed.
Thank you for your time!

**Comment by shashwat70** [comment]
@romani please review, Sir.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java]: I do not see code example after default config

Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java]: I thought it was really trivial, self-explanatory and obvious that's why I skipped it. However, I've added it now.

**Pull request closed without merging**